bootstrap/kubeadm/api/v1beta1/kubeadm_types.go

```diff
@@ -584,10 +584,10 @@ type Discovery struct {
 type BootstrapTokenDiscovery struct {
```

Comment:
This object now has no required field, so there will be difference in behaviour between not specifying the object, and specifying `bootstrapToken: {}`.

This is generally not desirable.

Is it valid to have an empty `bootstrapToken`? In the current use case, which alternative field would be used instead?

Reply:
From code reading what I can see is that the both the scenario's are handled currently, If `boostStrapToken` is not set and its nil, its being initialized first.
Also we have necessary checks for respecting the already set values as well.
Reference: https://github.com/kubernetes-sigs/cluster-api/blob/3e6bc6aa4903eec6f2d4e41848001aa2d5689e18/bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go#L1242-L1284


Please correct If I missed anything @chrischdi @fabriziopandini

Reply:
I think we are ok

Discovery has two mutually exclusive options, FileDiscovery and TokenDiscovery.

If File discovery is set, TokenDiscovery must not be set
If File discovery is not set, TokenDiscovery is used, either with the values specified by the users or, if those value are missing, with defaults.

Considering this, there wont be any difference in behaviour between not specifying the object, and specifying empty bootstrapToken: in both cases you accept all the defaults for TokenDiscovery